packages/react-component-library/src/components/CheckboxE/partials/StyledCheckmark.tsx

```diff
@@ -14,9 +17,46 @@ export const StyledCheckmark = styled.div`
   border: 1px solid ${color('neutral', '200')};
   border-radius: 3px;
 
+  &::before {
+    content: '';
+    ${position('absolute', 0, 0, 0, 0)};
+    border-radius: 2px;
+  }
+
   &::after {
     content: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgICA8cGF0aCBkPSJNMCAwaDI0djI0SDB6IiBmaWxsPSJub25lIiAvPgogICAgPHBhdGggdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMywgNCkiIGQ9Ik04LjgyMiAwYzEuMDUtLjAyNyAxLjU3NCAxLjMwNC44MjcgMi4wOEw0LjI5OCA3Ljc0YS44My44MyAwIDAgMS0xLjIwNyAwTC4zNzYgNC44NTVDLS43NTIgMy43MTguOTE0IDEuOTU2IDEuOTkgMy4xNDlsMS41MDggMS41OTVjLjEwNS4xMS4yODguMTEuNDA3IDBsNC4xMy00LjM3QTEuMSAxLjEgMCAwIDEgOC44MjMgMHoiCiAgICAgICAgICBmaWxsPSIjRkZGIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiLz4KPC9zdmc+Cg==);
+    display: none;
     border-radius: 2px;
     color: ${color('neutral', 'white')};
   }
+
+  ${StyledCheckbox}:hover, ${StyledCheckbox}:active {
```

Comment:
Previously the `StyledCheckbox` styled were spread across three files (which was hard to follow, and they contained some contradictions).

I used this pattern to move them all into this file (using a parent element as a selector):

https://styled-components.com/docs/advanced#referring-to-other-components